packages/yoastseo/src/parse/build/build.js

```diff
@@ -17,6 +18,19 @@ import permanentFilters from "./private/alwaysFilterElements";
  */
 export default function build( htmlString, languageProcessor ) {
 	let tree = adapt( parseFragment( htmlString, { sourceCodeLocationInfo: true } ) );
-	tree = filterTree( tree, permanentFilters );
-	return tokenize( tree, languageProcessor );
+	/*
+	 * Filter out some content from the tree so that it can be correctly tokenized. We don't want to tokenize text in
+	 * between 'code' and 'script' tags, but we do want to take into account the length of those elements when calculating
+	 * sentence and token positions.
+	 */
+	tree = filterBeforeTokenizing( tree );
```

Comment:
I wonder what was the reason for still keeping the position information of `code` and `script` but not other elements like `blockquote` etc.? (the way it's implemented in [this PR](https://github.com/Yoast/wordpress-seo/pull/20186/files#diff-37e00f16d732907dd0a36136b36c1d08230a361b2505b3c1620d551931eb7112)) . If there is a specific reason for this difference, do you mind adding that doc to the code as well?